pkgs/development/libraries/flatpak/fix-fonts-icons.patch

```diff
@@ -0,0 +1,92 @@
+diff --git a/common/flatpak-run.c b/common/flatpak-run.c
+index 94ad013..5c9f55e 100644
+--- a/common/flatpak-run.c
++++ b/common/flatpak-run.c
+@@ -871,6 +871,54 @@ out:
+   return res;
+ }
+ 
++static void
++get_nix_closure (GHashTable *closure, const gchar *source_path)
++{
++  if (g_file_test (source_path, G_FILE_TEST_IS_SYMLINK))
++    {
++      gchar *path = g_malloc(PATH_MAX);
++      realpath(source_path, path);
++      if (g_str_has_prefix(path, "/nix/store/"))
++        {
++          *strchr(path + strlen("/nix/store/"), '/') = 0;
++          g_hash_table_add(closure, path);
++        }
++      else
++        {
++          g_free(path);
++        }
++    }
++  else if (g_file_test (source_path, G_FILE_TEST_IS_DIR))
++    {
++      GDir *dir = g_dir_open(source_path, 0, NULL);
```

Comment:
This should be freed. You can have it done [automatically](https://github.com/GNOME/glib/blob/4144341e7a8a86c5eb980a0093e35c7606b9bff5/glib/glib-autocleanups.h#L55) with:

```suggestion
+      g_autoptr(GDir) dir = g_dir_open(source_path, 0, NULL);
```